consul/src/main/scala/io/buoyant/consul/v1/HealthStatus.scala

```diff
@@ -0,0 +1,45 @@
+package io.buoyant.consul.v1
+
+import com.fasterxml.jackson.core.{JsonGenerator, JsonParser}
+import com.fasterxml.jackson.databind.{DeserializationContext, SerializerProvider}
+import io.buoyant.config.{ConfigDeserializer, ConfigSerializer}
+
+object HealthStatus extends Enumeration {
```

Comment:
I'm not against using `Enumeration` and a custom deserializer here, but just for reference, I think we've generally represented JSON enums using a sealed class and `JsonSubTypes`, as in: 
https://github.com/linkerd/linkerd/blob/a9c1627d389fab206faac44d1f674695207ff5cd/namerd/storage/k8s/src/main/scala/io/buoyant/namerd/storage/kubernetes/DtabWatch.scala#L7-L23

Reply:
I was mostly copying from the consistency mode enum also in consul, only I used enum instead of a bunch of case classes so that they get an implicit ordering for the worst case method. It seems like there are also a bunch of custom deserialisers in the [config types package](https://github.com/linkerd/linkerd/tree/a9c1627d389fab206faac44d1f674695207ff5cd/config/src/main/scala/io/buoyant/config/types). Happy to change to a case class with a value or a JsonSubType etc if it's preferable though.

https://github.com/linkerd/linkerd/blob/a9c1627d389fab206faac44d1f674695207ff5cd/consul/src/main/scala/io/buoyant/consul/v1/ConsistencyMode.scala#L9-L29

Reply:
I think Enumeration is probably the right call in this case, just wanted to put that out there.